website/src/components/Toc.vue

```diff
@@ -26,11 +26,45 @@ export default {
     filteredHeadings() {
       return this.headings.filter(heading => heading.level < 4)
     }
+  },
+
+  watch: {
+    $route() {
+      this.isRoute = true
+      this.currentHash = this.$route.hash
+    }
+  },
+
+  data() {
+    return {
+      currentHash: null,
+      observer: null,
+      isRoute: false,
+      justMounted: true
+    }
+  },
+  
+  mounted() {
+    this.observer = new IntersectionObserver(([firstEntry]) => {
+      if (this.isRoute || this.justMounted) {
+      	this.isRoute = false
+      	this.justMounted = false
+      } else if (firstEntry.boundingClientRect.bottom <= firstEntry.intersectionRect.bottom) {
+        const hash = `#${firstEntry.target.id}`
+        history.pushState(null, null, hash)
+        this.currentHash = hash
```

Comment:
why not just `this.$router.push({hash})

Reply:
it change the scroll position. so will you are scrolling whenever you pass over a h tag it will return to it which make it looks buggy

Reply:
that's why I opened #269